pages/index.tsx

```diff
@@ -8,13 +8,18 @@ import LoginWithSMS from '../components/LoginWithSMS';
 import { LoginMethod } from '../lib/types';
 import LoginEntryPoint from '../components/LoginEntryPoint';
 
+const REDIRECT_URL_BASE = process.env.USE_VERCEL == 'true' ?
+  `https://${process.env.VERCEL_URL}/api/authenticate_magic_link` :
+  'http://localhost:3000/api/authenticate_magic_link';
```

Comment:
Key portion of the PR; this allows our example app to use Vercel's public URL, e.g. https://stytch-nextjs-integration-hm4as2csm-stytch-auth.vercel.app/, as the Redirect URL when deploying with Vercel and localhost when running locally.